packages/fast-components-react-base/src/horizontal-overflow/horizontal-overflow.tsx

```diff
@@ -0,0 +1,359 @@
+
+        if (children) {
+            const childNodes: HTMLElement[] = Array.prototype.slice.call(children);
+
+            for (const childNode of childNodes) {
+                const childNodeHeight: number = getClientRectWithMargin(childNode).height;
+
+                if (childNodeHeight > itemMaxHeight) {
+                    itemMaxHeight = childNodeHeight;
+                }
+            }
+        }
+
+        return itemMaxHeight;
+    }
+
+    /**
+     * Gets children by slot name
+     */
+    private getChildBySlot(slot: ButtonDirection): JSX.Element {
+        let childBySlot: JSX.Element = null;
+
+        React.Children.forEach(this.props.children, (child: JSX.Element): void => {
+            if (child.props && child.props.slot === slot) {
+                childBySlot = child;
+            }
+        });
+
+        return childBySlot;
+    }
+
+    /**
+     * Gets the children displayed is items to be scrolled
+     */
+    private getItems(): JSX.Element[] {
+        const previousButton: JSX.Element = this.getChildBySlot(ButtonDirection.previous);
+        const nextButton: JSX.Element = this.getChildBySlot(ButtonDirection.next);
+
+        return React.Children.map(this.props.children, (child: JSX.Element, index: number) => {
+            if (child !== nextButton && child !== previousButton) {
+                return (
+                    <li key={index} style={{display: "inline-block"}}>
+                        {child}
+                    </li>
+                );
+            }
+        });
+    }
+
+    /**
+     * Gets the distance to scroll based on the direction
+     */
+    private getScrollDistanceFromDirection(
+        direction: ButtonDirection,
+        availableWidth: number,
+        itemWidths: number[],
+        distanceFromBeginning: number
+    ): number {
+        if (itemWidths.length === 0) {
+            return 0;
+        }
+
+        let distance: number = 0;
+        const maxDistance: number = this.getMaxScrollDistance(availableWidth, itemWidths);
+
+        if (direction === ButtonDirection.next) {
+            distance = this.getWithinMaxDistance(distanceFromBeginning, availableWidth, itemWidths, maxDistance);
+        } else {
+            distance = this.getWithinMinDistance(distanceFromBeginning, availableWidth, itemWidths);
+        }
+
+        return distance;
+    }
+
+    /**
+     * Gets the distance unless it is over the maximum distance, then use maximum distance instead
+     */
+    private getWithinMaxDistance(distanceFromBeginning: number, availableWidth: number, itemWidths: number[], maxDistance: number): number {
+        let distance: number = 0;
+
+        if (distanceFromBeginning === maxDistance) {
+            distance = maxDistance;
+        } else {
+            distance = this.getNextDistance(availableWidth, itemWidths, distanceFromBeginning);
+
+            if (distance >= maxDistance) {
+                distance = maxDistance;
+            }
+        }
```

Comment:
With the else, it doesn't look like the original distance value ever gets returned without passing through the function on line 195. Could we simplify this a bit? I could be missing something.

```
if (distanceFromBeginning === maxDistance) {
    return maxDistance;
}

distance = this.getNextDistance(availableWidth, itemWidths, distanceFromBeginning);

return distance >= maxDistance ? maxDistance : distance;
```

Reply:
Sure, that looks much better